### PR TITLE
fix targets with zig toolchain

### PR DESCRIPTION
**Pull request by QuentinPerez**
When you use bazel + zig to cross compile a rust codebase, you got an 'invalid target error'
To fix the issue, a check has been added to know if it's a zig toolchain, not sure it's the right way to do it. 
Feel free to take over the PR.

### CODE REVIEW

**Review by NobodyXu — CHANGES_REQUESTED** [review]
It would be great if you could setup CI for testing zig-cc, to verify it indeed works as intended.

**Comment by QuentinPerez** [comment]
> It would be great if you could setup CI for testing zig-cc, to verify it indeed works as intended.

I tried it, but I am not sure how to achieve it correctly. Do you have any ideas or guidance on how I could do it?

**Comment by NobodyXu** [comment]
Simply install `pip3 install zig` in job test of `.github/workflows/main.yml`.

You would also need to add a `tests/zig-test.rs` and write all the testing for `zig-cc` in it.

File: src/tool.rs
Comment (NobodyXu) [inline src/tool.rs:100]: Just note that I've updated how compiler family detection work in #1000

Recommend you to wait for it to merge and then rebase against it.

**Review by NobodyXu — CHANGES_REQUESTED** [review]

**Comment by NobodyXu** [comment]
@QuentinPerez #1000 has been merged, can you rebase against main please?

File: src/lib.rs
Comment (NobodyXu) [inline src/lib.rs:1911]: Could do a

```suggestion
            let target = target.as_ref();

```

To avoid changing every if that uses target

**Review by NobodyXu — CHANGES_REQUESTED** [review]

**Review by NobodyXu — COMMENTED** [review]
Thanks, the code itself LGTM, can you add a test for zig-cc in CI?

You can just run cargo-test, while setting environment variable CC to zig-cc

**Comment by NobodyXu** [comment]
Discussed this with @thomcc on zulip, he mentioned that the zig-cc target triple might be changed in the later version of zig-cc.

We are afraid that after merging this PR, zig-cc might change its target triple and break this, causing more code to require to support both newer version and older version.

We decided to hold off adding more workarounds for zig-cc until it reaches 1.0, when its target triple and cmdline args are stable.

**Pull request closed without merging**
